lib/m2ts/caption-stream.js

```diff
@@ -264,9 +370,56 @@ var createDisplayBuffer = function() {
   return result;
 };
 
-var Cea608Stream = function() {
+var Cea608Stream = function(track) {
   Cea608Stream.prototype.init.call(this);
 
+  this.track_ = track || 1;
+  switch (this.track_) {
```

Comment:
I actually originally passed in `field` and `dataChannel`, but IIRC I changed it for a couple reasons. One was just that it looked nicer when constructing the objects (`this.ccn_ = new Cea608Stream(n);`), but more importantly, it's passed along to videojs-contrib-media-sources:

```javascript
this.trigger('data', {
  startPts: this.startPts_,
  endPts: pts,
  text: content,
  track: this.track_
});
```

I think that makes way more sense than passing the field and data channel. That being said, [videojs-contrib-media-sources is just using it as the track name](https://github.com/videojs/videojs-contrib-media-sources/pull/140/files#diff-cf1b209dfcf83384fb52b6a3c0eb5047R23). So maybe we could do something like this:

```javascript
var this.cc1_ = new Cea608Stream(0, 0);
...
var Cea608Stream = function(field, dataChannel) {
  // this next line is ugly, but should produce the correct track #
  this.name_ = 'cc' + (((field << 1) | dataChannel) + 1);
  this.field_ = field;
  this.dataChannel_ = dataChannel;
...
}
Cea608Stream.prototype.flushDisplayed = function(pts) {
...
  this.trigger('data', {
    startPts: this.startPts_,
    endPts: pts,
    text: content,
    track: this.name_
  });
}
```

Thoughts?